### PR TITLE
docs: mention the `inject` method in the DI guide

### PR DESCRIPTION
**Pull request by ToneyPK**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A
The `inject` method is not mentioned anywhere in the dependency injection - related guides.

## What is the new behavior?
Briefly mentioned the method and linked to it's documentation page.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/dependency-injection.md [hunk 1]
```diff
@@ -68,6 +68,15 @@ class HeroListComponent {
 }
 </code-example>
 
+Another way to inject a dependency without defining the constructor is to use the [inject](api/core/inject) method.
+
+<code-example language="typescript">
+@Component({ … })
+class HeroListComponent {
+  private service: inject(HeroService);
```
Comment (leo6104) [inline aio/content/guide/dependency-injection.md:76]:
```
private service = inject(HeroService);
```

Comment (ToneyPK) [reply aio/content/guide/dependency-injection.md]: good catch, thanks.
corrected

File: aio/content/guide/dependency-injection.md [hunk 2]
```diff
@@ -68,6 +68,15 @@ class HeroListComponent {
...
 </code-example>
 
+Another way to inject a dependency without defining the constructor is to use the [inject](api/core/inject) method.
+
+<code-example language="typescript">
+@Component({ … })
+class HeroListComponent {
+  private service = inject(HeroService);
+}
+</code-example>
```
Comment (AndrewKushnir) [inline aio/content/guide/dependency-injection.md:78]: @ToneyPK thanks for improving Angular docs!

I think this change would also benefit from a link to a guide that explains what an injection context is (i.e. where the `inject` function can be used). The mentioned guide is in a different PR, see #49782. Marking this PR as blocked for now (blocked by PR #49782).

Comment (ToneyPK) [reply aio/content/guide/dependency-injection.md:78]: @AndrewKushnir My pleasure.

Yes, even a better idea!

File: aio/content/guide/dependency-injection.md [hunk 3]
```diff
@@ -68,6 +68,15 @@ class HeroListComponent {
 }
 </code-example>
 
+Another way to inject a dependency without defining the constructor is to use the [inject](api/core/inject) method.
```
Comment (AndrewKushnir) [inline aio/content/guide/dependency-injection.md:71]:
```suggestion
Another option is to use the [inject](api/core/inject) method:
```

File: aio/content/guide/dependency-injection.md (hunk 2 above)
Comment (AndrewKushnir) [reply aio/content/guide/dependency-injection.md:78]: The mentioned PR is now merged and it also added a reference to the injection context into the `inject` docs, so we can omit it here. I've just left a minor comment on how we can simplify the wording a bit.

File: aio/content/guide/dependency-injection.md (hunk 3 above)
Comment (ToneyPK) [reply aio/content/guide/dependency-injection.md:71]: Is it fine if I just apply your suggestion (we end up with two commits unsquashed)? I don’t have a laptop until next week to squash the commits but I don’t want to halt the progress on the PR.

Comment (AndrewKushnir) [reply aio/content/guide/dependency-injection.md:71]: @ToneyPK yes, you can try doing this and use a commit message like this: `docs: simplify a note in DI guide` (so the `lint` CI job passes). Please let me know if that worked. Thank you.

Comment (ToneyPK) [reply aio/content/guide/dependency-injection.md]: @AndrewKushnir I applied the commit with your message and the lint job has passed successfully.

**Review by AndrewKushnir — APPROVED** [review]

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 385526239990e98a0b456109bbaf79227a57901a.
